### PR TITLE
Meteors No Longer Hurt Floors

### PR DESCRIPTION
**Pull request by Neerti**

### CODE REVIEW

**Comment by Neerti** [comment]
Not sure this is a bug.

**Comment by Belsima** [comment]
Does that mean they only blow up walls?

**Comment by Atermonera** [comment]
This is affecting the meteor itself, not the explosion it generates. So if a meteor hits deck 3 on SC, it won't leave a trail of broken/destroyed floors from intercept to explosion, it'll just be an explosion.

**Comment by Anewbe** [comment]
Should still punch through walls and stuff, but there won't be a million extra floor holes after a meteor hit.

**Comment by Leshana** [comment]
I think it will make meteors cooler, I support this change.

File: code/game/gamemodes/meteor/meteors.dm [hunk 1]
```diff
@@ -153,16 +153,17 @@
 /obj/effect/meteor/proc/ram_turf(var/turf/T)
 	//first bust whatever is in the turf
 	for(var/atom/A in T)
-		if(A != src)
-			A.ex_act(hitpwr)
+		if(A == src) // Don't hit ourselves.
+			continue
+		if(isturf(A)) // Don't hit floors. We'll deal with walls later.
```
Comment (Leshana) [inline code/game/gamemodes/meteor/meteors.dm]: I'm fairly sure that turfs do not contain themselves or any other turfs, so this if condition will never be true.

**Comment by Belsima** [comment]
Ah, good stuff then.

**Pull request merged**